### PR TITLE
TST verify return none inplace in tests/indexing

### PR DESCRIPTION
**Pull request by r-toroxel**
verify we return none for all inplace calls in tests/indexing

related: https://github.com/pandas-dev/pandas/pull/35210

### CODE REVIEW

**Comment by r-toroxel** [comment]
**build failed**, it says

ci/run_tests.sh

ImportError while loading conftest '/home/travis/build/pandas-dev/pandas/pandas/conftest.py'.

ImportError: C extension: 'dtypes' from partially initialized module 'pandas._libs.tslibs' (most likely due to a circular import) (/home/travis/build/pandas-dev/pandas/pandas/_libs/tslibs/__init__.py) not built. If you want to import pandas from the source directory, you may need to run 'python setup.py build_ext --inplace --force' to build the C extensions first.

The command "ci/run_tests.sh" exited with 4.

**Comment by r-toroxel** [comment]
had a similar issue locally and had to recompile everything ...

File: pandas/tests/indexing/multiindex/test_chaining_and_caching.py [hunk 1]
```diff
@@ -25,7 +25,8 @@ def test_detect_chained_assignment():
 
     msg = "A value is trying to be set on a copy of a slice from a DataFrame"
     with pytest.raises(com.SettingWithCopyError, match=msg):
-        zed["eyes"]["right"].fillna(value=555, inplace=True)
+        return_value = zed["eyes"]["right"].fillna(value=555, inplace=True)
+        assert return_value is None
```
Comment (r-toroxel) [inline pandas/tests/indexing/multiindex/test_chaining_and_caching.py]: fixed this one

File: pandas/tests/frame/test_constructors.py [hunk 2]
```diff
@@ -1178,6 +1178,13 @@ def test_constructor_list_of_odicts(self):
         expected = DataFrame(index=[0])
         tm.assert_frame_equal(result, expected)
 
+    def test_constructor_single_row(self):
+        data = [OrderedDict([["a", 1.5], ["b", 3], ["c", 4], ["d", 6]])]
+
+        result = DataFrame(data)
+        expected = DataFrame.from_dict(dict(zip([0], data)), orient="index")
+        tm.assert_frame_equal(result, expected.reindex(result.index))
```
Comment (mroeschke) [inline pandas/tests/frame/test_constructors.py:1186]: Nit: Could you put the reindex in the `expected = ...` statement

**Comment by simonjayhawkins** [comment]
@r-toroxel Thanks for the PR. The branch has conflicts with master. can you merge upstream/master

**Review by mroeschke — APPROVED** [review]

**Comment by jreback** [comment]
thanks @r-toroxel

**Pull request merged**